io.sloeber.ui/src/io/sloeber/ui/monitor/internal/SerialListener.java

```diff
@@ -122,16 +125,49 @@ public void dispose() {
 	public class TxtUpdater implements Runnable {
 		private boolean running = false;
 		private String additionalSerialData = new String();
+		private StringBuilder lineBuffer = new StringBuilder();
+
+		private DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm:ss"); //$NON-NLS-1$
+		private static final String arduinoEOL = "\r\n"; //$NON-NLS-1$
+		private final int eolLength = arduinoEOL.length();
 
 		public synchronized void addData(String event) {
 			if (!event.isEmpty()) {
-				this.additionalSerialData = this.additionalSerialData + event;
```

Comment:
wouldn't it be easier to have something like
```
if (SerialListener.showTimestamps) {
	String ts = LocalTime.now().format(timeFormatter) + ": "; //$NON-NLS-1$
        this.additionalSerialData = this.additionalSerialData + event.replace('\n',ts );
}
else{
         this.additionalSerialData = this.additionalSerialData + event;
}
```

Reply:
I tried that but it wasn't reliable. I can't tell you why because it looks like it should work but it didn't. I still got line breaks in weird places, it was like they were being added by something else but I can't think what. It made no sense, but it didn't work.

The code in the pull request is the only one of 3 or 4 different algorithms I tried that seems to work consistently.

Reply:
Thinking about this.... I think it is way better to add this code in the monitor itself
SerialMonitor.ReportSerialActivity
go from
	public void ReportSerialActivity(String stInfo, int style) {
		int startPoint = monitorOutput.getCharCount();
		monitorOutput.append(stInfo);
		StyleRange styleRange = new StyleRange();
		styleRange.start = startPoint;
               styleRange.length = stInfo.length();

to
	public void ReportSerialActivity(String stInfo, int style) {
		int startPoint = monitorOutput.getCharCount();
                String displayInfo;
if (SerialListener.showTimestamps) {
	String ts = LocalTime.now().format(timeFormatter) + ": "; //$NON-NLS-1$
        displayInfo= stInfo+ event.replace('\n',ts );
}
else{
         displayInfo= stInfo;
}

		monitorOutput.append(displayInfo);
		StyleRange styleRange = new StyleRange();
		styleRange.start = startPoint;
                styleRange.length = displayInfo.length();